lib/parse/client.rb

```diff
@@ -69,6 +69,11 @@ def request(uri, method = :get, body = nil, query = nil, content_type = nil)
       end
 
       @session.send(method, uri, query || body || {}, headers).body
+
+    # Don't leak our internal libraries to our clients.
+    # Extend this list of exceptions as needed.
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError
+      raise Parse::ConnectionError
```

Comment:
@henrik done, I also pushed a commit to pass along the actual error message. https://github.com/adelevie/parse-ruby-client/commit/72c3881ef0b94b45e1311b30a6bb4c2ac66e44d0